src/IB/IBFESurfaceMethod.cpp

```diff
@@ -1216,37 +1843,82 @@ IBFESurfaceMethod::initializeFEEquationSystems()
             {
                 std::ostringstream os;
                 os << "U_" << d;
-                U_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
+					U_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
             }
 
             System& U_n_system = equation_systems->add_system<System>(NORMAL_VELOCITY_SYSTEM_NAME);
             for (unsigned int d = 0; d < NDIM; ++d)
             {
                 std::ostringstream os;
                 os << "U_n_" << d;
-                U_n_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
+					U_n_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
             }
 
             System& U_t_system = equation_systems->add_system<System>(TANGENTIAL_VELOCITY_SYSTEM_NAME);
             for (unsigned int d = 0; d < NDIM; ++d)
             {
                 std::ostringstream os;
                 os << "U_t_" << d;
-                U_t_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
+					U_t_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
             }
 
             System& F_system = equation_systems->add_system<System>(FORCE_SYSTEM_NAME);
             for (unsigned int d = 0; d < NDIM; ++d)
             {
                 std::ostringstream os;
                 os << "F_" << d;
-                F_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
+					F_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
             }
 
-            if (d_use_jump_conditions)
+            if (d_use_pressure_jump_conditions)
             {
-                System& DP_system = equation_systems->add_system<System>(PRESSURE_JUMP_SYSTEM_NAME);
-                DP_system.add_variable("C_p", d_fe_order[part], d_fe_family[part]);
+                System& P_jump_system = equation_systems->add_system<System>(PRESSURE_JUMP_SYSTEM_NAME);
+                if (d_use_l2_lagrange_family)
+                    P_jump_system.add_variable("P_jump_", d_fe_order[part], L2_LAGRANGE);
+                else
+                    P_jump_system.add_variable("P_jump_", d_fe_order[part], d_fe_family[part]);
+
+                System& P_system = equation_systems->add_system<System>(P_SYSTEM_NAME);
+                P_system.add_variable("P_", d_fe_order[part], d_fe_family[part]);
+            }
+
+            if (d_use_velocity_jump_conditions)
+            {
+                boost::array<System*, NDIM> DU_jump_system;
+                for (unsigned int d = 0; d < NDIM; ++d)
+                {
+                    DU_jump_system[d] = &equation_systems->add_system<System>(VELOCITY_JUMP_SYSTEM_NAME[d]);
+
+                    for (unsigned int i = 0; i < NDIM; ++i)
+                    {
+                        std::ostringstream os;
+                        os << "DU_jump_" << d << "_" << i;
+                        if (d_use_l2_lagrange_family)
```

Comment:
I would suggest to keep track of the `FEFamily`s used for different types of quantities (displacements/velocities, forces, stresses/jump conditions, etc) instead of using this kind of an `if`-`else` construction.